### PR TITLE
Remove cmdlinet from init/lazy_goto_model

### PR DESCRIPTION
**Pull request by peterschrammel**
It should be possible to build a goto model without requiring a cmdlinet, e.g. in unit tests.

Caution! This breaks most driver programs.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [X] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: e2d75fa889117ce94cc09417e92caaed6fca5ce1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/93300370
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: jbmc/src/jbmc/jbmc_parse_options.cpp [hunk 1]
```diff
@@ -106,8 +106,11 @@ void jbmc_parse_optionst::get_command_line_options(optionst &options)
     usage_error();
     exit(1); // should contemplate EX_USAGE from sysexits.h
   }
-
```
Comment (tautschnig) [inline jbmc/src/jbmc/jbmc_parse_options.cpp]: Nit pick: unrelated change?

File: jbmc/unit/pointer-analysis/custom_value_set_analysis.cpp [hunk 2]
```diff
@@ -172,20 +172,19 @@ SCENARIO("test_value_set_analysis",
...
     // is global and previous unit tests may have altered it
-    command_line.set("java-cp-include-files", "CustomVSATest.class");
     config.java.classpath={"."};
-    command_line.args.push_back("pointer-analysis/CustomVSATest.jar");
 
-    register_language(new_java_bytecode_language);
     optionst options;
-    parse_java_language_options(command_line, options);
+    options.set_option("java-cp-include-files", "CustomVSATest.class");
+    std::vector<std::string> files = {"pointer-analysis/CustomVSATest.jar"};
```
Comment (tautschnig) [inline jbmc/unit/pointer-analysis/custom_value_set_analysis.cpp]: Nit pick: it may make things easier to read if just passing this directly to `initialize_goto_model` instead of adding this `files` local variable. Just less context for the reader to remember.

File: src/langapi/language_ui.cpp [hunk 3]
```diff
@@ -112,8 +113,12 @@ bool language_uit::final()
 {
   language_files.set_message_handler(*message_handler);
 
+  // TODO: This should be moved elsewhere because it is not used in
+  //       this repository.
```
Comment (tautschnig) [inline src/langapi/language_ui.cpp:117]: Maybe create an issue for this?

**Review by tautschnig — APPROVED** [review]
A nice step forward! I am assuming TG just needs a genuine update due to the API change.

**Comment by peterschrammel** [comment]
>  I am assuming TG just needs a genuine update due to the API change.

Already got one that is passing.

File: src/langapi/language_ui.cpp (hunk 3 above)
Comment (peterschrammel) [reply src/langapi/language_ui.cpp:117]: https://github.com/diffblue/cbmc/issues/3519

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 7183e66b9cc9fb5f82bf84a7263a29109c4e1c0b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/93331192
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by smowton — APPROVED** [review]

File: src/langapi/language_ui.cpp [hunk 4]
```diff
@@ -112,8 +113,12 @@ bool language_uit::final()
...
   // Enable/disable stub generation for opaque methods
-  bool stubs_enabled=_cmdline.isset("generate-opaque-stubs");
+  bool stubs_enabled = false;
+  if(options != nullptr)
+    stubs_enabled = options->is_set("generate-opaque-stubs");
```
Comment (chrisr-diffblue) [inline src/langapi/language_ui.cpp:121]: Is this any easier to read if written as `bool stubs_enabled = options != nullptr ? options->is_set(..) : false;` ?

**Review by chrisr-diffblue — APPROVED** [review]
Looks like a nice cleanup.

File: src/langapi/language_ui.cpp (hunk 4 above)
Comment (peterschrammel) [reply src/langapi/language_ui.cpp:121]: Will be removed in #3520 anyway.
